eventlet/greenio/base.py

```diff
@@ -145,7 +145,7 @@ def __init__(self, family=socket.AF_INET, *args, **kwargs):
         except AttributeError:
             self._timeout = socket.getdefaulttimeout()
 
-        if should_set_nonblocking:
+        if should_set_nonblocking and fd.fileno() != -1:
```

Comment:
A comment explaining the second part of the condition would be helpful here.

Reply:
Done